### PR TITLE
MINOR: Using lambdas in producer package

### PR DESCRIPTION
**Pull request by 17hao**
Maybe using lambda will be more concise.
I replaced old codes with lambdas in `org/apache/kafka/clients/producer` and `org/apache/kafka/clients/producer/internals` packages.

### CODE REVIEW

**Review by rondagostino — COMMENTED** [review]
Thanks for the PR.  Can you use `for i in $(git grep -l "new [a-zA-Z0-9]*() {$"); do grep -H "new [a-zA-Z0-9]*() {$" $i ; done` to look for other potential changes?  If you are feeling inspired you can try `for i in $(git grep -l "new [a-zA-Z0-9]*(.*) {$"); do grep -H "new [a-zA-Z0-9]*(.*) {$" $i ; done` (which triples the number of possibilities)

File: clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java [hunk 1]
```diff
@@ -208,12 +208,9 @@ private void maybeRemoveAndDeallocateBatch(ProducerBatch batch) {
 
     private void addToInflightBatches(List<ProducerBatch> batches) {
         for (ProducerBatch batch : batches) {
-            List<ProducerBatch> inflightBatchList = inFlightBatches.get(batch.topicPartition);
-            if (inflightBatchList == null) {
-                inflightBatchList = new ArrayList<>();
-                inFlightBatches.put(batch.topicPartition, inflightBatchList);
-            }
-            inflightBatchList.add(batch);
+            Objects.requireNonNull(inFlightBatches.computeIfAbsent(batch.topicPartition,
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java:211]: Why do we need `requireNonNull` here?

Comment (17hao) [reply clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java:211]: Sorry I didn't think over it. `requireNonNull` is redundant.

**Comment by 17hao** [comment]
I began to doubt is it worth for stylistic only changes but didn't bring significant benefits? 
Anyway thanks for reviewed it @rondagostino @ijuma !

**Pull request closed without merging**